libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1341,9 +1341,13 @@ bool TypeChecker::visit(IndexAccess const& _access)
 		else
 		{
 			expectType(*index, IntegerType(256));
-			if (auto integerType = dynamic_cast<IntegerConstantType const*>(type(*index).get()))
-				if (!actualType.isDynamicallySized() && actualType.length() <= integerType->literalValue(nullptr))
+			if (auto numberType = dynamic_cast<RationalNumberType const*>(type(*index).get()))
+			{
+				if (numberType->denominator() != 1)
```

Comment:
You can change that to an assert - we already checked for implicit convertibility to `u256` in line 1343.